src/prefect/logging/handlers.py

```diff
@@ -51,11 +52,15 @@ def _min_interval(self):
     async def _handle_batch(self, items: List):
         try:
             await self._client.create_logs(items)
-        except Exception:
+        except Exception as e:
             # Roughly replicate the behavior of the stdlib logger error handling
             if logging.raiseExceptions and sys.stderr:
                 sys.stderr.write("--- Error logging to API ---\n")
-                traceback.print_exc(file=sys.stderr)
+                if PREFECT_LOGGING_INTERNAL_LEVEL.value() == "DEBUG":
+                    traceback.print_exc(file=sys.stderr)
+                else:
+                    # Only log the exception message in non-DEBUG mode
+                    sys.stderr.write(f"{e}")
```

Comment:
Very small nit, but using `str` seems cleaner to me.

```suggestion
                    sys.stderr.write(str(e))
```

Reply:
I like it!